internal/server/v1/dlq/handler_test.go

```diff
@@ -0,0 +1,231 @@
+package dlq
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	entropyv1beta1rpc "buf.build/gen/go/gotocompany/proton/grpc/go/gotocompany/entropy/v1beta1/entropyv1beta1grpc"
+	entropyv1beta1 "buf.build/gen/go/gotocompany/proton/protocolbuffers/go/gotocompany/entropy/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+
+	"github.com/goto/dex/entropy"
+	"github.com/goto/dex/internal/server/gcs"
+	"github.com/goto/dex/internal/server/utils"
+	"github.com/goto/dex/internal/server/v1/firehose"
+)
+
+type mockEntropyService struct {
+	mock.Mock
+}
+
+func (*mockEntropyService) ListResources(context.Context, *entropyv1beta1.ListResourcesRequest, ...grpc.CallOption) (*entropyv1beta1.ListResourcesResponse, error) {
+	panic("")
+}
+
+func (e *mockEntropyService) GetResource(ctx context.Context, in *entropyv1beta1.GetResourceRequest, opts ...grpc.CallOption) (*entropyv1beta1.GetResourceResponse, error) {
+	args := e.Called(ctx, in, opts)
+	if args.Get(0) == nil {
+		err, _ := args.Get(1).(error)
+		return nil, err
+	}
+	response, _ := args.Get(0).(*entropyv1beta1.GetResourceResponse)
+	return response, nil
+}
+
+func (*mockEntropyService) CreateResource(context.Context, *entropyv1beta1.CreateResourceRequest, ...grpc.CallOption) (*entropyv1beta1.CreateResourceResponse, error) {
+	panic("")
+}
+
+func (*mockEntropyService) UpdateResource(context.Context, *entropyv1beta1.UpdateResourceRequest, ...grpc.CallOption) (*entropyv1beta1.UpdateResourceResponse, error) {
+	panic("")
+}
+
+func (*mockEntropyService) DeleteResource(context.Context, *entropyv1beta1.DeleteResourceRequest, ...grpc.CallOption) (*entropyv1beta1.DeleteResourceResponse, error) {
+	panic("")
+}
+
+func (*mockEntropyService) ApplyAction(context.Context, *entropyv1beta1.ApplyActionRequest, ...grpc.CallOption) (*entropyv1beta1.ApplyActionResponse, error) {
+	panic("")
+}
+
+func (*mockEntropyService) GetLog(context.Context, *entropyv1beta1.GetLogRequest, ...grpc.CallOption) (entropyv1beta1rpc.ResourceService_GetLogClient, error) {
+	panic("")
+}
+
+func (*mockEntropyService) GetResourceRevisions(context.Context, *entropyv1beta1.GetResourceRevisionsRequest, ...grpc.CallOption) (*entropyv1beta1.GetResourceRevisionsResponse, error) {
+	panic("")
+}
+
+type mockGcsClient struct {
+	mock.Mock
+}
+
+func (g *mockGcsClient) ListTopicDates(bucketInfo gcs.BucketInfo) (map[string]map[string]int64, error) {
+	args := g.Called(bucketInfo)
+	if args.Get(0) == nil {
+		err, _ := args.Get(1).(error)
+		return nil, err
+	}
+	topicDates, _ := args.Get(0).(map[string]map[string]int64)
+	return topicDates, nil
+}
+
+type mockHTTPWriter struct {
```

Comment:
do we need to mock http writer? for most cases we should be able to use the one from golang package itself.

example: https://github.com/goto/dex/blob/main/internal/server/v1/alert/routes_test.go#L55

Reply:
this is fine, will rename to testHttpRequest. we are no running a mock server.